src/IBM.Cloud.SDK.Core/Authentication/CloudPakForData/CloudPakForDataAuthenticator.cs

```
@@ -171,9 +278,9 @@ public override void Validate()
                 throw new ArgumentNullException(string.Format(ErrorMessagePropMissing, "Username"));
             }
 
-            if (string.IsNullOrEmpty(Password))
+            if (string.IsNullOrEmpty(Password) || string.IsNullOrEmpty(Apikey))
             {
-                throw new ArgumentNullException(string.Format(ErrorMessagePropMissing, "Password"));
+                throw new ArgumentNullException("The Password or Apikey property is required but was not specified.");

```

Should we throw this error message in the base class? 
https://github.com/IBM/dotnet-sdk-core/blob/main/src/IBM.Cloud.SDK.Core/Authentication/Authenticator.cs#L46-L48

---

Initial thought I was thinking about updating the Authenticator file with the new message as well. But this error message is really specific for this case, so I decided to just update as raw message here.

For this case, we want the users to provide either `password` or `apikey`. So I'm not sure if we should create a template like `{0} or {1} is required ....` or just use raw message here.
 
Please let me know if which way that our team prefer.

---

I vote for the template, just to keep it consistent with other authenticators

---

Maybe you can use
```
throw new ArgumentNullException(string.Format(ErrorMessagePropMissing, "password or apikey"));
```

>The password or apikey property is required but was not specified.

---

Ah. Smart, I did not think if it before! Will update it. Thanks